api/service/stagedstreamsync/stage_epoch.go

```diff
@@ -34,6 +35,9 @@ func NewStageEpochCfg(bc core.BlockChain, db kv.RwDB) StageEpochCfg {
 
 func (sr *StageEpoch) Exec(ctx context.Context, firstCycle bool, invalidBlockRevert bool, s *StageState, reverter Reverter, tx kv.RwTx) error {
 
+	s.state.Debug("shard", sr.configs.bc.ShardID())
```

Comment:
Any issue updating 
```golang
func (fi *fastIterator) Debug() 
```
to use `utils.Logger().Debug()` ?

Reply:
Agreed. I have been working on sync issues for the last few weeks, and it was quite challenging to trace and replicate the errors locally. To address this, I added a debug mode, which was extremely useful.
The debug mode is a hidden option located under sync options in the config file. It is optional and, by default, set to false, meaning it doesn't affect the normal execution. Even if you mistakenly set this flag incorrectly, it won't cause any issues; it exists only for display purposes. The sync debug mode allows printing out the variables and each individual stage's process, helping to trace the issues and implementation correctness.
I'm considering removing all the debug lines and retaining this function for future use, but I'm not entirely sure about it yet.